lib/registry-secrets.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * @license
+ * Copyright 2018 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This module contains interfaces and classes for the validation of the JSON
+ * format expected by the balena builder and the docker daemon for the
+ * authentication of private docker registries.
+ */
+
+import * as ajv from 'ajv';
+
+import { RegistrySecretValidationError } from './errors';
+export { RegistrySecretValidationError } from './errors';
```

Comment:
Not sure if this import/export pair is common, but when combined with this other line in `index.ts`:
```
export * from './registry-secrets';
```
It allows me to write the following in other repos like `balena-cli`:
```
import {
    RegistrySecretValidator,
    RegistrySecretValidationError,
} from 'resin-multibuild';
```